src/Microsoft.Health.Fhir.TemplateManagement/Factory/TemplateCollectionProviderFactory.cs

```diff
@@ -59,16 +59,23 @@ public ITemplateCollectionProvider CreateTemplateCollectionProvider(string image
             return new TemplateCollectionProvider(imageInfo, client, _templateCache, _configuration);
         }
 
-        public void InitDefaultTemplates()
+        private void InitDefaultTemplates()
         {
-            TemplateLayer defaultTemplateLayer = TemplateLayer.ReadFromEmbeddedResource();
-            _templateCache.Set(ImageInfo.DefaultTemplateImageReference, defaultTemplateLayer, new MemoryCacheEntryOptions() { AbsoluteExpiration = ObjectCache.InfiniteAbsoluteExpiration, Size = defaultTemplateLayer.Size, Priority = Extensions.Caching.Memory.CacheItemPriority.NeverRemove });
-            _templateCache.Set(defaultTemplateLayer.Digest, defaultTemplateLayer, new MemoryCacheEntryOptions() { AbsoluteExpiration = ObjectCache.InfiniteAbsoluteExpiration, Size = defaultTemplateLayer.Size, Priority = Extensions.Caching.Memory.CacheItemPriority.NeverRemove });
+            foreach (var templateInfo in Constants.DefaultTemplateInfo)
+            {
+                TemplateLayer templateLayer = TemplateLayer.ReadFromEmbeddedResource(templateInfo.Value.Item1);
+                _templateCache.Set(templateInfo.Value.Item2, templateLayer, new MemoryCacheEntryOptions() { AbsoluteExpiration = ObjectCache.InfiniteAbsoluteExpiration, Size = templateLayer.Size, Priority = Extensions.Caching.Memory.CacheItemPriority.NeverRemove });
+                _templateCache.Set(templateLayer.Digest, templateLayer, new MemoryCacheEntryOptions() { AbsoluteExpiration = ObjectCache.InfiniteAbsoluteExpiration, Size = templateLayer.Size, Priority = Extensions.Caching.Memory.CacheItemPriority.NeverRemove });
+                if (templateInfo.Key == Constants.DefaultDataType)
```

Comment:
why do we need this default type?

Reply:
In order to be compatible with the previous version, in addition to use "microsofthealth/hl7v2templates:default" for hl7v2 and "microsofthealth/ccdatemplates:default" for ccda, we also use "microsofthealth/fhirconverter:default" for default hl7v2 templates, so we also need to add it to template cache here for datatype hl7v2. The "DefaultDataType" is used for this purpose.

Reply:
Could we move defaultImageReference "microsofthealth/fhirconverter:default" into DefaultImageInfo's DefaultTemplateMap? The Reference name could be the key.
Seems we don't need this default type.

Reply:
The key of DefaultTemplateMap is DataType, it couldn't add  "microsofthealth/fhirconverter:default" to DefaultTemplateMap due to key duplication. The default type is only used if the datatype is null to keep compatible with the previous, I could remove it and use DataType.Hl7v2 directly instead.

Reply:
So why not using image reference as key to avoid conflict?

Reply:
resolve as suggestion.